docs/utxo-hd/v0.1/for-developers.md

```diff
@@ -0,0 +1,276 @@
+# UTxO-HD for developers
+
+This document aims to provide a guide on how to integrate the UTxO-HD
+implementation into downstream clients.
+
+Clarification requests and questions are more than welcome.
+
+## Why UTxO-HD
+
+Over time, the `cardano-node` implementation's memory usage has been steadily
+increasing. A sync with mainnet shows that currently the node uses around 16GBs
+of memory at the peak. Not every user of the Cardano blockchain has a computer
+with that much memory and therefore it becomes necessary to lower the memory of
+the running node.
+
+A big chunk of that memory is consumed by mantaining the UTxO set, with its many
+millions of entries. UTxO-HD solves this by extracting this set from the ledger
+state and dumping it into the disk. This effectively lowers the memory by 5GB
+(tentative numbers shown by ad-hoc benchmarks, should not be considered exact).
+However this comes at a cost, both in speed performance and in complexity of the
+code.
+
+This design is based on the technical report "Storing the Cardano ledger state
+on disk: analysis and design options" by Duncan Coutts and Douglas Wilson.
+
+## Implementation
+
+The Cardano Ledger code remains unaltered, oblivious to the fact that the
+Consensus layer is managing the UTxOs present in a ledger state, but by design,
+the Ledger rules that involve evaluating a block on top of a ledger state only
+need to consult the UTxOs that are consumed by that block, and produce UTxOs
+that were not present in the UTxO set of the input state. This allows us to
+provide ledger states to the ledger rules in which the UTxO set is just a
+subset/restriction to the values required by the block in turn.
+
```

Comment:
Here we should link the idea of "hydrating the ledger state" to the *need for diffs*.


... we use the idea of anti-cache